### ISSUE TITLE
0.16.3 has issues when parsing double or np.float64 on Windows platforms

### ISSUE THREAD

**Issue by An-Li-GitHub** [tebeka/fastavro#125]
Hey guys,

i did not have this issue while using 0.16.1 nor 0.14.11.

code to replicate

envrionment: python 2.7.13
cython: 0.25.2
fastavro: 0.16.3

============================================================
import fastavro as avro
from fastavro import writer

schema = {
    'doc': 'A weather reading.',
    'name': 'Weather',
    'namespace': 'test',
    'type': 'record',
    'fields': [
        {'name': 'station', 'type': 'string'},
        {'name': 'time', 'type': 'long'},
        {'name': 'temp', 'type': 'int'},
        {'name': 'test_float', 'type': 'double'}
    ],
}

with open('weather_out.avro', 'wb') as out:
    writer(out, schema, [{u'station': u'011990-99999',
  u'temp': 0L,
  'test_float': 0.21334215134123513,
  u'time': -714214260L},
 {u'station': u'011990-99999',
  u'temp': 22L,
  'test_float': 0.21334215134123513,
  u'time': -714213259L},
 {u'station': u'011990-99999',
  u'temp': -11L,
  'test_float': 0.21334215134123513,
  u'time': -714210269L},
 {u'station': u'012650-99999',
  u'temp': 111L,
  'test_float': 0.21334215134123513,
  u'time': -714208170L}])

with open('weather_out.avro', 'rb') as fo:
    reader = avro.reader(fo)
    schema = reader.schema
    
    for x in reader:
        print x

**Comment by scottbelden** [comment]
Hmm, do you mind expanding on what is wrong? I ran your code and it appeared to work. It looks like the output matches the input (unless I didn't look close enough). This is what I got...

```python
In [5]: %paste
import fastavro as avro
from fastavro import writer

schema = {
'doc': 'A weather reading.',
'name': 'Weather',
'namespace': 'test',
'type': 'record',
'fields': [
{'name': 'station', 'type': 'string'},
{'name': 'time', 'type': 'long'},
{'name': 'temp', 'type': 'int'},
{'name': 'test_float', 'type': 'double'}
],
}

with open('weather_out.avro', 'wb') as out:
    writer(out, schema, [{u'station': u'011990-99999',
    u'temp': 0L,
    'test_float': 0.21334215134123513,
    u'time': -714214260L},
    {u'station': u'011990-99999',
    u'temp': 22L,
    'test_float': 0.21334215134123513,
    u'time': -714213259L},
    {u'station': u'011990-99999',
    u'temp': -11L,
    'test_float': 0.21334215134123513,
    u'time': -714210269L},
    {u'station': u'012650-99999',
    u'temp': 111L,
    'test_float': 0.21334215134123513,
    u'time': -714208170L}])

with open('weather_out.avro', 'rb') as fo:
    reader = avro.reader(fo)
    schema = reader.schema
    for x in reader:
        print x

## -- End pasted text --
{u'test_float': 0.21334215134123513, u'station': u'011990-99999', u'temp': 0, u'time': -714214260}
{u'test_float': 0.21334215134123513, u'station': u'011990-99999', u'temp': 22, u'time': -714213259}
{u'test_float': 0.21334215134123513, u'station': u'011990-99999', u'temp': -11, u'time': -714210269}
{u'test_float': 0.21334215134123513, u'station': u'012650-99999', u'temp': 111, u'time': -714208170}
```

**Comment by An-Li-GitHub** [comment]
hmm that's odd. what i got is
{u'test_float': 1.437606092e-314, u'station': u'011990-99999', u'temp': 0, u'time': -714214260}
{u'test_float': 1.437606092e-314, u'station': u'011990-99999', u'temp': 22, u'time': -714213259}
{u'test_float': 1.437606092e-314, u'station': u'011990-99999', u'temp': -11, u'time': -714210269}
{u'test_float': 1.437606092e-314, u'station': u'012650-99999', u'temp': 111, u'time': -714208170}

if i change the schema of "test_float" from "double" to "float". i got  
{u'test_float': 0.21334214508533478, u'station': u'011990-99999', u'temp': 0, u'time': -714214260}
{u'test_float': 0.21334214508533478, u'station': u'011990-99999', u'temp': 22, u'time': -714213259}
{u'test_float': 0.21334214508533478, u'station': u'011990-99999', u'temp': -11, u'time': -714210269}
{u'test_float': 0.21334214508533478, u'station': u'012650-99999', u'temp': 111, u'time': -714208170}

if i use 0.14.11 instead of 0.16.3, i got the the same result as yours.
can you share what package that you are using? pip list or conda list would be nice.
and i tried a few ways to install 0.16.3, pip, conda, get the repo and build local. results are the same.
mine: 

===
backports-abc (0.5)
backports.shutil-get-terminal-size (1.0.0)
backports.ssl-match-hostname (3.5.0.1)
bleach (1.5.0)
cdecimal (2.3)
colorama (0.3.9)
configparser (3.5.0)
Cython (0.25.2)
decorator (4.1.2)
entrypoints (0.2.3)
enum34 (1.1.6)
fastavro (0.16.3)
functools32 (3.2.3.post2)
html5lib (0.999)
ipykernel (4.6.1)
ipython (5.3.0)
ipython-genutils (0.2.0)
ipywidgets (6.0.0)
Jinja2 (2.9.6)
jsonschema (2.6.0)
jupyter (1.0.0)
jupyter-client (5.1.0)
jupyter-console (5.2.0)
jupyter-core (4.3.0)
MarkupSafe (1.0)
mistune (0.7.4)
nbconvert (5.2.1)
nbformat (4.4.0)
notebook (5.0.0)
numpy (1.12.1)
pandas (0.19.2)
pandocfilters (1.4.2)
path.py (10.3.1)
pathlib2 (2.3.0)
pickleshare (0.7.4)
pip (9.0.1)
prompt-toolkit (1.0.15)
Pygments (2.2.0)
pyodbc (3.0.10)
python-dateutil (2.6.0)
pytz (2017.2)
pyzmq (16.0.2)
qtconsole (4.2.1)
redis (2.10.5)
scandir (1.5)
setuptools (27.2.0)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.10.0)
testpath (0.3)
tornado (4.5.2)
traitlets (4.3.2)
wcwidth (0.1.7)
wheel (0.29.0)
widgetsnbextension (3.0.2)
win-unicode-console (0.5)

**Comment by scottbelden** [comment]
If I change from `double` to `float` I get the same result as you, but when I use `double` it still roundtrips correctly. My py2 is `2.7.10` (so not the latest like yours). My pip list is the following:

```
appnope (0.1.0)
attrs (17.3.0)
backports.shutil-get-terminal-size (1.0.0)
check-manifest (0.36)
configparser (3.5.0)
Cython (0.25.2)
decorator (4.1.2)
enum34 (1.1.6)
fastavro (0.16.3)
flake8 (3.5.0)
funcsigs (1.0.2)
ipython (5.5.0)
ipython-genutils (0.2.0)
mccabe (0.6.1)
pathlib2 (2.3.0)
pexpect (4.3.0)
pickleshare (0.7.4)
pip (9.0.1)
pluggy (0.6.0)
prompt-toolkit (1.0.15)
ptyprocess (0.5.2)
py (1.5.2)
pycodestyle (2.3.1)
pyflakes (1.6.0)
Pygments (2.2.0)
pytest (3.3.0)
scandir (1.6)
setuptools (38.2.3)
simplegeneric (0.8.1)
six (1.11.0)
traitlets (4.3.2)
wcwidth (0.1.7)
wheel (0.30.0)
```

I'll make a branch with the test case so that travis can check it (since the py2 travis is 2.7.13).

**Comment by scottbelden** [comment]
I make a branch with the test case: https://github.com/tebeka/fastavro/compare/issue_125

When it ran on travis it failed for any py3 version (because of the `L`s which was removed on py3, but it passed on 2.7.13: https://travis-ci.org/tebeka/fastavro/builds/310017873?utm_source=github_status&utm_medium=notification

**Comment by An-Li-GitHub** [comment]
Tried on a linux EC2 and it worked fine. Tried on 2 different windows boxes and the result are consistently wrong. I think it has something to do with OS. Does this version have anything that platform specific?

**Comment by scottbelden** [comment]
Yeah, it does seem to be windows specific. I also noticed that this seems to be specific to the compiled cython files. For example, in my test environment my fastavro install directory has `.pyd` files on windows and if I delete/move those so that only the pure python files are used I get the correct round trip. I created an issue to add windows testing (https://github.com/tebeka/fastavro/issues/126) but for the moment we might need to push a temporary fix so that windows only uses the pure python files.

**Comment by scottbelden** [comment]
@An-Li-GitHub Could you see if https://github.com/tebeka/fastavro/commit/f746eb6e38a6ec73c38eae7c22fc50061de50abc fixes the issue you are seeing? It looks like it does on a Windows box I created, but I don't have automated Windows testing working yet, so it's hard to confirm for sure.

**Comment by An-Li-GitHub** [comment]
@scottbelden sorry for the late response. yeah that would do it. nice job thanks. when would there be a new version patched with this? we will stick with 0.14.11 for now until this fix is in.

Thanks again

**Comment by scottbelden** [comment]
@An-Li-GitHub The new version should be up on pypi. It will be available on conda-forge shortly as well.
